src/uu/fmt/src/linebreak.rs

```diff
@@ -367,14 +375,19 @@ fn find_kp_breakpoints<'a, T: Iterator<Item = &'a WordInfo<'a>>>(
     build_best_path(&linebreaks, active_breaks)
 }
 
-fn build_best_path<'a>(paths: &[LineBreak<'a>], active: &[usize]) -> Vec<(&'a WordInfo<'a>, bool)> {
+fn build_best_path<'a>(
+    paths: &[LineBreak<'a>],
+    active: &[usize],
+) -> UResult<Vec<(&'a WordInfo<'a>, bool)>> {
     let mut breakwords = vec![];
     // of the active paths, we select the one with the fewest demerits
     let mut best_idx = match active.iter().min_by_key(|&&a| paths[a].demerits) {
-        None => crash!(
-            1,
-            "Failed to find a k-p linebreak solution. This should never happen."
-        ),
+        None => {
+            return Err(USimpleError::new(
+                1,
+                "Failed to find a k-p linebreak solution. This should never happen.",
```

Comment:
I think this text is right actually, this can, as far as I can tell, never happen. This is because the `min_by_key` call only returns `None` if the slice is empty, but the `active` slice is initialized with at least one value. Maybe an `unreachable!` call is more appropriate here. Although, I would also love to see the code refactored so that this case is eliminated. For example, we could also choose to just `unwrap_or(0)` on this `min_by_key` call, because `active` is initialized with `0`. I'll leave that decision up to you :)

Reply:
This makes sense. Will mark the arm `unreachable!` as `unwrap_or` can give the impression that the default value can possibly be picked.

Reply:
Refactored to remove it completely.